source/corvusoft/restbed/detail/socket_impl.cpp

```diff
@@ -159,6 +159,7 @@ namespace restbed
         void SocketImpl::write( const Bytes& data, const function< void ( const error_code&, size_t ) >& callback )
         {
             m_buffer = make_shared< Bytes >( data );
+            if (m_is_open) {
```

Comment:
What if m_is_open is false ? Wouldn't that cause the callback to never be called ?